src/executor.rs

```diff
@@ -265,6 +287,22 @@ where
     }
 }
 
+/// Policy to deal with CMP and SUB instructions
+pub enum CmpPolicy {
```

Comment:
This doesn't sound correct to me. Cmplog and Cmpcov can be used together, though might be not too effective. It should be bitflags in my opinion.

Reply:
It's what the C-version of unicornafl does🤔<https://github.com/AFLplusplus/unicornafl/blob/a8f96daacacd2d49212b3e9f08eee3c459b4aee1/unicornafl.cpp#L587>

Reply:
And qemuafl also makes it exclusive: <https://github.com/AFLplusplus/qemuafl/blob/c43dd6e0369cd5d2a2458f3bd7f4f58c8de53300/qemuafl/cpu-translate.h#L79>

Reply:
I agree. I'll check the implementation later.

Reply:
@vanhauser-thc what’s the rationale of that? And what if the cmplog binary emits different coverage? Like fuzzing binary has cmpcov while cmplog binary doesn’t have.

Reply:
the coverage the cmplog binary produces is not important, just that it does, otherwise the afl-fuzz forkserver will complain

Reply:
I think so - try it :-)

Reply:
I think similar optimisation could apply to afl-cc too. Will have a look either.

Reply:
Shall we first merge this PR and let the random-bitmap-set approach be another PR?

Reply:
@vanhauser-thc Sorry to ping you. I tried to totally remove edge coverage feedback for cmplog binaries in unicornafl, and everything just works fine. The AFL++ is the latest dev branch with default compilation setup (which defines `CMPLOG_COMBINE`). I checked the code, and it seems that the only location that requires coverage seems to be [here](https://github.com/AFLplusplus/AFLplusplus/blob/6b1d6a905525a136d0ab4da709e05d8a94401a47/src/afl-fuzz-redqueen.c#L3203-L3236), which looks fine to me if coverage is empty. Would you mind taking a look at the related afl++ code when you are free? Thank you!

Reply:
The variable you referred is all about the main fsrv of the binary.

I assume maybe we can build cmplog binariy with this: https://github.com/AFLplusplus/AFLplusplus/pull/2421

Reply:
Hmmm I read this PR, if I understand it right, does this mean that we just don't mind whether instrumentation when cmplog, instead we just obey what the environment said. And this problem should be handled by AFL++ side?

Reply:
> @vanhauser-thc Sorry to ping you. I tried to totally remove edge coverage feedback for cmplog binaries in unicornafl, and everything just works fine. The AFL++ is the latest dev branch with default compilation setup (which defines `CMPLOG_COMBINE`). I checked the code, and it seems that the only location that requires coverage seems to be [here](https://github.com/AFLplusplus/AFLplusplus/blob/6b1d6a905525a136d0ab4da709e05d8a94401a47/src/afl-fuzz-redqueen.c#L3203-L3236), which looks fine to me if coverage is empty. Would you mind taking a look at the related afl++ code when you are free? Thank you!

I looked and yes, the cmplog run does not need coverage, but the forkserver needs the one dummy byte set

Reply:
I can update to set a random by for the `AFL_LLVM/GCC`_ONLY_FSRV`.

From my personal experience, remove pc instrumentation has 10-20% speedup =)

Reply:
> Hmmm I read this PR, if I understand it right, does this mean that we just don't mind whether instrumentation when cmplog, instead we just obey what the environment said. And this problem should be handled by AFL++ side?

Sorry missed this message. I was talking about general fuzzing.

Reply:
@wtdcode The byte is already set [here](https://github.com/AFLplusplus/unicornafl/blob/b85f4fec1dd25abac404727fe1090b02546dc867/src/harness.rs#L53-L56)

Reply:
> @wtdcode The byte is already set [here](https://github.com/AFLplusplus/unicornafl/blob/b85f4fec1dd25abac404727fe1090b02546dc867/src/harness.rs#L53-L56)

Yeah, I understand. Sorry I should not mix the thread here.

I mean here: https://github.com/AFLplusplus/AFLplusplus/blob/stable/instrumentation/README.cmplog.md#build

Now maybe we can also have `export AFL_LLVM_ONLY_FSRV=1` ? @vanhauser-thc

Reply:
I have pushed a new commit, which disables PC instrumentation for CMPLOG mode :)

Reply:
But I need to update the docs to recommend doing this

Reply:
How about going to docs?

Reply:
Okay, make sense. I was going to refer to https://github.com/AFLplusplus/AFLplusplus/blob/stable/docs/fuzzing_in_depth.md